http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamApiClientProvider.scala

```diff
@@ -26,16 +26,26 @@ trait SamApiClientProvider[F[_]] {
   def azureApi(token: String)(implicit ev: Ask[F, AppContext]): F[AzureApi]
 }
 
-class HttpSamApiClientProvider[F[_]](samUrl: String)(implicit F: Async[F]) extends SamApiClientProvider[F] {
-  private val okHttpClient = new ApiClient().getHttpClient
+class HttpSamApiClientProvider[F[_]](samUrl: String, maxConcurrentRequests: Int)(implicit F: Async[F])
+    extends SamApiClientProvider[F] {
+  private val okHttpClient = buildOkHttpClient
   private val timeout = 30 seconds
 
+  private def buildOkHttpClient = {
+    val dispatcher = new Dispatcher()
+    dispatcher.setMaxRequests(maxConcurrentRequests)
+    dispatcher.setMaxRequestsPerHost(maxConcurrentRequests)
```

Comment:
this client is only handling a single host so these are both set to the same value